src/bin/main.rs

```diff
@@ -84,6 +85,17 @@ fn main() -> Result<()> {
         )
     })?;
 
+    // Guard against running as root. This check can be overridden by changing `allow_root` inside
+    // the config file.
+    let allow_root = config.core_settings.allow_root.unwrap_or(false);
+    if get_current_uid() == 0 && !allow_root {
```

Comment:
You can swap these two so the `get_current_uid` isn't triggered if `allow_root` is true